File: src/main/java/com/vielheit/core/domain/Entry.java
package com.vielheit.core.domain;

import javax.persistence.*;

/**
 * jcm - 4/16/17.
 */
@Entity
@Table(name = "Entry", schema = "vielheit")
public class Entry {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;
}

File: src/main/kotlin/cc/kojeve/vielheit/domain/Entry.kt
package cc.kojeve.vielheit.domain

import javax.persistence.*

@Entity
class Entry(
        @ManyToOne
        @JoinColumn(name = "user_id")
        var user: User,
        @Column
        val title: String,
        @ManyToMany(mappedBy = "entries", cascade = [CascadeType.ALL])
        val tags: MutableSet<Tag> = mutableSetOf(),
        @ManyToMany(mappedBy = "entries", cascade = [CascadeType.ALL])
        val relations: MutableSet<Relation> = mutableSetOf()
) : Domain()